include/alpaka/core/ConcurrentExecPool.hpp

```diff
@@ -33,41 +33,37 @@ namespace alpaka::core
     namespace detail
     {
         template<typename T>
-        class ThreadSafeQueue : private std::queue<T>
+        struct ThreadSafeQueue
         {
-        public:
             ThreadSafeQueue() = default;
-            //! \return If the queue is empty.
+
             [[nodiscard]] auto empty() const -> bool
             {
-                return std::queue<T>::empty();
+                return m_queue.empty();
             }
+
             //! Pushes the given value onto the back of the queue.
-            auto push(T&& t) -> void
+            void push(T&& t)
             {
-                std::lock_guard<std::mutex> lk(m_Mutex);
-
-                std::queue<T>::push(std::forward<T>(t));
+                std::lock_guard<std::mutex> lk(m_mutex);
```

Comment:
Is the `<std::mutex>` part still needed with c++17 ?

Reply:
It shoudn't and I would prefer CTAD here, but clang gave me a warning (`-Wctad-maybe-unsupported`) so I provided the type explicitely.